src/modules/roc_address/io_uri.h

```diff
@@ -12,37 +12,56 @@
 #ifndef ROC_ADDRESS_IO_URI_H_
 #define ROC_ADDRESS_IO_URI_H_
 
+#include "roc_core/array.h"
+#include "roc_core/noncopyable.h"
 #include "roc_core/stddefs.h"
 
 namespace roc {
 namespace address {
 
 //! Audio file or device URI.
-struct IoURI {
+class IoURI : public core::NonCopyable<> {
+public:
     //! Initialize empty URI.
-    IoURI();
+    IoURI(core::IAllocator&);
 
-    //! Returns true if the URI is empty.
-    bool is_empty() const;
+    //! Returns true if the URI has all required components.
```

Comment:
Could you specify what these components are?

Reply:
Done.